javav2/example_code/firehose/src/main/java/com/example/firehose/scenario/FirehoseScenario.java

```diff
@@ -0,0 +1,232 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.firehose.scenario;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import software.amazon.awssdk.services.cloudwatch.model.*;
+import software.amazon.awssdk.services.firehose.FirehoseClient;
+import software.amazon.awssdk.services.firehose.model.PutRecordBatchRequest;
+import software.amazon.awssdk.services.firehose.model.PutRecordBatchResponse;
+import software.amazon.awssdk.services.firehose.model.PutRecordRequest;
+import software.amazon.awssdk.services.firehose.model.Record;
+import software.amazon.awssdk.regions.Region;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Before running this Java V2 code example, set up your development
+ * environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+
+// snippet-start:[firehose.java2.scenario.main]
+public class FirehoseScenario {
+    private static FirehoseClient firehoseClient;
+    private static CloudWatchClient cloudWatchClient;
+    private static String deliveryStreamName;
+
+    public static void main(String[] args) {
+        firehoseClient = FirehoseClient.builder().region(Region.US_EAST_1).build();
+        cloudWatchClient = CloudWatchClient.builder().region(Region.US_EAST_1).build();
+
+        // Replace with your region and delivery stream name
+        deliveryStreamName = "stream35";
```

Comment:
Wouldn't it be better to pass this as a program argument?

Reply:
fixed